Library/Homebrew/brew.sh

```diff
@@ -114,6 +114,25 @@ else
   : "${HOMEBREW_OS_VERSION:=$(uname -r)}"
   HOMEBREW_OS_USER_AGENT_VERSION="$HOMEBREW_OS_VERSION"
 
+  # Ensure the system Curl is a version that supports modern HTTPS certificates.
+  HOMEBREW_SYSTEM_CURL_VERSION="$($(command -v curl) --version 2>/dev/null | head -n1 | awk '{print $2}')"
+  # shellcheck disable=SC2086,SC2183
+  printf -v HOMEBREW_SYSTEM_CURL_VERSION_NUMERIC "%02d%02d%02d" ${HOMEBREW_SYSTEM_CURL_VERSION//./ }
+  if [[ "10#$HOMEBREW_SYSTEM_CURL_VERSION_NUMERIC" -lt "10#074100" ]]
+  then
+    HOMEBREW_SYSTEM_CURL_TOO_OLD="1"
+    HOMEBREW_FORCE_BREWED_CURL="1"
+  fi
+
+  # Ensure the system Git is at or newer than the version set in diagnostic.rb.
+  HOMEBREW_SYSTEM_GIT_VERSION="$($(command -v git) --version 2>/dev/null | awk '{print $3}')"
+  # shellcheck disable=SC2086,SC2183
+  printf -v HOMEBREW_SYSTEM_GIT_VERSION_NUMERIC "%02d%02d%02d%02d" ${HOMEBREW_SYSTEM_GIT_VERSION//./ }
+  if [[ "10#$HOMEBREW_SYSTEM_GIT_VERSION_NUMERIC" -lt "10#02070000" ]]
```

Comment:
Ensures the numbers are compared using base 10. Otherwise the leading zero makes them read as octal.

Reply:
Can the leading zeros be stripped/avoided instead?

Reply:
Another thought: rather than manually keeping this in sync ideally the logic in `diagnostic.rb` would use something from here instead of having versions hardcoded in both places.

Reply:
`HOMEBREW_SYSTEM_GIT_TOO_OLD` or similar, perhaps.

Reply:
Easily. Note to self: revisit this when `git` hits version 10.

Reply:
I think it's a very reasonable assumption that Git and Curl are both always >=1.0.0